### PR TITLE
Upgrade all Dynamo projects to .net 4.6.1.

### PR DESCRIPTION
**Pull request by ikeough**
### Purpose

This PR updates all Dynamo projects in the main solution to target .net framework version 4.6.1. This will enable us to author Dynamo libraries against .net standard 2.0. Those libraries can then be used with dotnet core 2.0 projects on linux, mac, etc. In order to make .net standard libraries work, the `netstandard.dll` must exist in Dynamo's root directory. The `NETStandard.Library.NETFramework` nuget package has been added to `DynamoCore`, and the require prebuild copy step has been added for the `netstandard.dll`.

**This will require an update of associated Nuget packages.**

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers
@mjkkirschner @QilongTang @saintentropy

### CODE REVIEW

File: src/Engine/ProtoAssociative/ProtoAssociative.csproj [hunk 1]
```diff
@@ -9,7 +9,7 @@
     <OutputType>Library</OutputType>
     <RootNamespace>ProtoAssociative</RootNamespace>
     <AssemblyName>ProtoAssociative</AssemblyName>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.7</TargetFrameworkVersion>
```
Comment (QilongTang) [inline src/Engine/ProtoAssociative/ProtoAssociative.csproj]: Any reason why this one is different than others?

Comment (ikeough) [reply src/Engine/ProtoAssociative/ProtoAssociative.csproj]: Hmmm. Shouldn't be. I'll check.

**Comment by QilongTang** [comment]
@ikeough I was trying to run Unit Test on your branch using EC and saw build errors in log https://ecwest.autodesk.com/commander/link/workspaceFile/workspaces/SCL-Workspace?jobStepId=3496beaa-b03c-11e7-9133-a0d3c1eff5a4&fileName=Build+Dynamo.3496beaa-b03c-11e7-9133-a0d3c1eff5a4.log&jobName=Dynamo_DevCI_Self_Service_Dynamo_9009173_201710131030&jobId=34757adf-b03c-11e7-bff0-a0d3c1eff5a4&diagCharEncoding=&resourceName=USECSDDWINCI04&completed=1, not sure if it's BRE machine not setup with target .Net or something else. I will pull your branch locally and take a look.

**Comment by mjkkirschner** [comment]
besides updates to the nuget packages-  what other consequences will there be for the ecosystem of Dynamo packages, sample repos, and other client implementations? I think this work is very valuable, but before we merge it in we should try to size the other associated work.

**Comment by QilongTang** [comment]
@mjkkirschner @ikeough I run through Dynamo Unit tests against this branch locally and did not find too many surprises. Less worried about tests now but what are the `associated work` you mentioned?

**Comment by ramramps** [comment]
what about the build systems (Engops)? Does it conflict with D4R?

**Comment by mjkkirschner** [comment]
I'm alright with the mono build failing since we would move to use .net core as well... or even better... don't do anything.

I'm asking the same questions as Ram above, D4R, sample repo, core packages like MeshToolKit etc etc etc.

**Comment by ramramps** [comment]
well, if Reach is no more, then why do we need mono build? we can just use .net core and move on...

**Comment by mjkkirschner** [comment]
@ramramps ❗️ exactly.

**Comment by mjkkirschner** [comment]
@QilongTang are we doing this?

**Comment by jnealb** [comment]
If we are, please do it soon to allow for testing time.

**Comment by mjkkirschner** [comment]
ping.

**Comment by QilongTang** [comment]
Closing for now.

**Pull request closed without merging**